<a href="https://colab.research.google.com/github/andresmartinezcel2510-debug/Nootbook_agro2026_1/blob/main/Trabajo_individual_Situaciones_Ingenier%C3%ADas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Ejercicio 1: Capacidad de Silos de Granos

### 1. ANÁLISIS DEL PROBLEMA (Modelo E-P-S)
**Entradas:**
*   radio [float]: Radio del silo en metros.
*   altura [float]: Altura del silo en metros.

**Proceso:**
1.  Calcular el volumen del silo cilíndrico usando la fórmula:

$$Volumen = π * radio^2 * altura$$

2.  Calcular las toneladas de maíz que puede almacenar multiplicando el volumen por la densidad del maíz:
$$Toneladas = Volumen * 0.75$$

**Salida:**
*   toneladas_maiz [float]: Cantidad de toneladas de maíz que puede almacenar el silo.



### 2. CÓDIGO FUENTE EN PYTHON

In [7]:
import math

# Se solicitan las entradas convirtiendo a punto flotante (decimal)
radio = float(input("Ingrese el radio del silo en metros: "))
altura = float(input("Ingrese la altura del silo en metros: "))

# Datos conocidos
densidad_maiz = 0.75 # ton/m3

# Cálculo matemático del volumen del silo (V = π * r^2 * h)
volumen_silo = math.pi * (radio ** 2) * altura

# Cálculo de las toneladas de maíz
toneladas_maiz = volumen_silo * densidad_maiz

# Salida de información formateada
print(f"Un silo con radio {radio}m y altura {altura}m puede almacenar aproximadamente {toneladas_maiz:.2f} toneladas de maíz.")

Ingrese el radio del silo en metros: 3
Ingrese la altura del silo en metros: 10
Un silo con radio 3.0m y altura 10.0m puede almacenar aproximadamente 212.06 toneladas de maíz.


### 3. DOCUMENTACIÓN TÉCNICA (Explicación de líneas)
*   **Línea 1:** import math importa el módulo math para poder usar la constante math.pi (el valor de Pi).

*   **Línea 4:**
`radio = float(input("Ingrese el radio del silo en metros: "))` solicita al usuario el radio del silo. **input()** captura la entrada como texto, y **float()** la convierte a un número de punto flotante.

*   **Línea 5:**  
`altura = float(input("Ingrese la altura del silo en metros: "))` solicita al usuario la altura del silo, convirtiéndola también a un número de punto flotante.
*   **Línea 8:**  
`densidad_maiz` = 0.75 define una variable para almacenar la densidad del maíz (0.75 toneladas por metro cúbico).
*   **Línea 11:**  
`volumen_silo = math.pi * (radio ** 2) * altura` calcula el volumen del silo usando la fórmula `π * r^2 * h`. `** 2` es el operador de exponenciación para elevar al cuadrado.
*   **Línea 14:**  
`toneladas_maiz = volumen_silo * densidad_maiz` calcula la cantidad total de maíz en toneladas multiplicando el volumen por la densidad.
*   **Línea 17:**  
`print(f"...")` muestra el resultado final. Se utiliza una f-string (cadena formateada) para insertar los valores de las variables directamente en el texto. `:.2f` formatea el número a dos decimales.

### 4. PRUEBA DE ESCRITORIO
| Variable         | Valor Ingresado | Resultado del Cálculo                 |
| :--------------- | :-------------- | :------------------------------------ |
| `radio`          | 3.0             |                                       |
| `altura`         | 10.0            |                                       |
| `densidad_maiz`  |                 | 0.75                                  |
| `volumen_silo`   |                 | `math.pi * (3^2) * 10` ≈ `3.14159 * 9 * 10` = 282.743 |
| **toneladasmaiz** |                 | `282.743 * 0.75` ≈ 212.057             |

## Ejercicio 2: Dosificación de Fertilizante Líquido

### 1. ANÁLISIS DEL PROBLEMA (Modelo E-P-S)
**Entradas:**
*   `hectareas_totales` [float]: Número total de hectáreas a regar.
*   `dosis litros por hectarea` [float]: Dosis de fertilizante requerida por hectárea en litros.

**Proceso:**
1.  Calcular la cantidad total de producto fertilizante necesaria:

`Producto_total = hectareas_totales * dosis_litros_por_hec`

2.  Calcular la cantidad total de agua necesaria para la dilución, basándose en la relación 1:500: `Agua_total = Producto_total * 500`.

**Salida:**
*   `litros_producto` [float]: Cantidad total de producto fertilizante necesario en litros.
*   `litros_agua` [float]: Cantidad total de agua necesaria en litros.



### 2. CÓDIGO FUENTE EN PYTHON

In [8]:
# Se solicitan las entradas convirtiendo a punto flotante (decimal)
hectareas_totales = float(input("Ingrese el total de hectáreas: "))
dosis_litros_por_ha = float(input("Ingrese la dosis requerida (litros/hectárea): "))

# Datos conocidos
relacion_dilucion = 500 # 1 litro de producto por 500 litros de agua

# Cálculo del total de producto fertilizante necesario
litros_producto = hectareas_totales * dosis_litros_por_ha

# Cálculo del total de agua necesario para la dilución
litros_agua = litros_producto * relacion_dilucion

# Salida de información formateada
print(f"Para {hectareas_totales:.0f} hectáreas con una dosis de {dosis_litros_por_ha:.0f} L/ha:")
print(f"Se necesitan {litros_producto:.0f} litros de producto fertilizante.")
print(f"Se necesitan {litros_agua:.0f} litros de agua total para la dilución.")

Ingrese el total de hectáreas: 5
Ingrese la dosis requerida (litros/hectárea): 2
Para 5 hectáreas con una dosis de 2 L/ha:
Se necesitan 10 litros de producto fertilizante.
Se necesitan 5000 litros de agua total para la dilución.


### 3. DOCUMENTACIÓN TÉCNICA (Explicación de líneas)
*   **Línea 2:** `hectareas_totales = float(input("Ingrese el total de hectáreas: "))` solicita al usuario el número de hectáreas y lo convierte a un número de punto flotante.
*   **Línea 3:** `dosis_litros_por_ha = float(input("Ingrese la dosis requerida (litros/hectárea): "))` solicita la dosis de fertilizante por hectárea y la convierte a float.
*   **Línea 6:** `relacion_dilucion = 500` define una variable para la relación de dilución (500 litros de agua por cada litro de producto).
*   **Línea 9:** `litros_producto = hectareas_totales * dosis_litros_por_ha` calcula la cantidad total de producto fertilizante requerida.
*   **Línea 12:** `litros_agua = litros_producto * relacion_dilucion` calcula la cantidad total de agua necesaria para diluir el producto.
*   **Líneas 15-17:** `print(f"...")` muestran los resultados formateados usando f-strings. `:.0f` formatea el número a cero decimales, es decir, como un entero.

### 4. PRUEBA DE ESCRITORIO
| Variable            | Valor Ingresado | Resultado del Cálculo                 |
| :------------------ | :-------------- | :------------------------------------ |
| `hectareas_totales` | 5.0             |                                       |
| `dosis_litros_por_ha`| 2.0             |                                       |
| `relacion_dilucion` |                 | 500                                   |
| `litros_producto`   |                 | `5 * 2` = 10.0                        |
| `litros_agua`       |                 | `10 * 500` = 5000.0                   |

## Ejercicio 3: Rendimiento de Pulpa en Frutas

### 1. ANÁLISIS DEL PROBLEMA (Modelo E-P-S)
**Entradas:**
*   `peso_total_fruta` [float]: Peso total de la fruta comprada en kg.
*   `peso_residuos` [float]: Peso de los residuos (cáscaras y semillas) en kg.

**Proceso:**
1.  Calcular el peso de la pulpa: `Pulpa = Peso Total - Residuos`.
2.  Calcular el porcentaje de rendimiento: `Porcentaje_Rendimiento = (Pulpa / Peso Total) * 100`.

**Salida:**
*   `peso_pulpa` [float]: Peso de la pulpa obtenida en kg.
*   `porcentaje_rendimiento` [float]: Porcentaje de rendimiento de la pulpa.



### 2. CÓDIGO FUENTE EN PYTHON

In [9]:
# Se solicitan las entradas convirtiendo a punto flotante (decimal)
peso_total_fruta = float(input("Ingrese el peso total de la fruta comprada en kg: "))
peso_residuos = float(input("Ingrese el peso de los residuos (cáscaras y semillas) en kg: "))

# Cálculo del peso de la pulpa
peso_pulpa = peso_total_fruta - peso_residuos

# Cálculo del porcentaje de rendimiento
# Se verifica que peso_total_fruta no sea cero para evitar división por cero
if peso_total_fruta > 0:
    porcentaje_rendimiento = (peso_pulpa / peso_total_fruta) * 100
else:
    porcentaje_rendimiento = 0.0 # Si no hay fruta, el rendimiento es 0
    print("Advertencia: El peso total de la fruta es cero, el rendimiento se establece en 0%.")

# Salida de información formateada
print(f"Peso de la pulpa obtenida: {peso_pulpa:.2f} kg.")
print(f"Porcentaje de rendimiento: {porcentaje_rendimiento:.2f} %.")

Ingrese el peso total de la fruta comprada en kg: 500
Ingrese el peso de los residuos (cáscaras y semillas) en kg: 150
Peso de la pulpa obtenida: 350.00 kg.
Porcentaje de rendimiento: 70.00 %.


### 3. DOCUMENTACIÓN TÉCNICA (Explicación de líneas)
*   **Línea 2:** `peso_total_fruta = float(input("Ingrese el peso total de la fruta comprada en kg: "))` solicita el peso total de la fruta y lo convierte a un número de punto flotante.
*   **Línea 3:** `peso_residuos = float(input("Ingrese el peso de los residuos (cáscaras y semillas) en kg: "))` solicita el peso de los residuos y lo convierte a float.
*   **Línea 6:** `peso_pulpa = peso_total_fruta - peso_residuos` calcula el peso de la pulpa restando los residuos al peso total.
*   **Línea 10:** `if peso_total_fruta > 0:` Inicia un bloque condicional para verificar si el peso total de la fruta es mayor que cero. Esto previene un error de división por cero si el usuario ingresa 0 para el peso total.
*   **Línea 11:** `porcentaje_rendimiento = (peso_pulpa / peso_total_fruta) * 100` calcula el porcentaje de rendimiento si el peso total es positivo.
*   **Línea 12:** `else:` Define el bloque que se ejecuta si la condición anterior es falsa (es decir, si `peso_total_fruta` es 0 o negativo).
*   **Línea 13:** `porcentaje_rendimiento = 0.0` Asigna 0% de rendimiento si no hay fruta.
*   **Línea 14:** `print("Advertencia: ...")` Muestra una advertencia al usuario en caso de que el peso total de la fruta sea cero.
*   **Líneas 17-18:** `print(f"...")` muestran los resultados formateados (peso de la pulpa y porcentaje de rendimiento) usando f-strings, con dos decimales.

### 4. PRUEBA DE ESCRITORIO
| Variable               | Valor Ingresado | Resultado del Cálculo                 |
| :--------------------- | :-------------- | :------------------------------------ |
| `peso_total_fruta`     | 500.0           |                                       |
| `peso_residuos`        | 150.0           |                                       |
| `peso_pulpa`           |                 | `500 - 150` = 350.0                   |
| `porcentaje_rendimiento` |                 | `(350 / 500) * 100` = 70.0            |